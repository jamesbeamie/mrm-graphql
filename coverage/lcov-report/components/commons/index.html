<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/commons</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> components/commons
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">17.71% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>48/271</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">7.59% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>11/145</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">10.14% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>7/69</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.68% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>48/257</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="ActionButtons.js"><a href="ActionButtons.js.html">ActionButtons.js</a></td>
	<td data-value="75" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 75%;"></div><div class="cover-empty" style="width:25%;"></div></div></td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	</tr>

<tr>
	<td class="file low" data-value="BackArrow.js"><a href="BackArrow.js.html">BackArrow.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	</tr>

<tr>
	<td class="file high" data-value="Button.js"><a href="Button.js.html">Button.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file low" data-value="Calendar.js"><a href="Calendar.js.html">Calendar.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	</tr>

<tr>
	<td class="file low" data-value="CheckboxSlide.js"><a href="CheckboxSlide.js.html">CheckboxSlide.js</a></td>
	<td data-value="25" class="pic low"><div class="chart"><div class="cover-fill" style="width: 25%;"></div><div class="cover-empty" style="width:75%;"></div></div></td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="32" class="abs low">8/32</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="26.67" class="pct low">26.67%</td>
	<td data-value="30" class="abs low">8/30</td>
	</tr>

<tr>
	<td class="file medium" data-value="DataNotFound.js"><a href="DataNotFound.js.html">DataNotFound.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	</tr>

<tr>
	<td class="file low" data-value="Dropdown.js"><a href="Dropdown.js.html">Dropdown.js</a></td>
	<td data-value="12.5" class="pic low"><div class="chart"><div class="cover-fill" style="width: 12%;"></div><div class="cover-empty" style="width:88%;"></div></div></td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="8" class="abs low">1/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="8" class="abs low">1/8</td>
	</tr>

<tr>
	<td class="file low" data-value="EpicTowerInputs.js"><a href="EpicTowerInputs.js.html">EpicTowerInputs.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="ErrorBoundary.js"><a href="ErrorBoundary.js.html">ErrorBoundary.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	</tr>

<tr>
	<td class="file high" data-value="ErrorIcon.js"><a href="ErrorIcon.js.html">ErrorIcon.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file low" data-value="ExportButton.js"><a href="ExportButton.js.html">ExportButton.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	</tr>

<tr>
	<td class="file low" data-value="IconButtons.js"><a href="IconButtons.js.html">IconButtons.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	</tr>

<tr>
	<td class="file low" data-value="Input.js"><a href="Input.js.html">Input.js</a></td>
	<td data-value="11.76" class="pic low"><div class="chart"><div class="cover-fill" style="width: 11%;"></div><div class="cover-empty" style="width:89%;"></div></div></td>
	<td data-value="11.76" class="pct low">11.76%</td>
	<td data-value="17" class="abs low">2/17</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="11.76" class="pct low">11.76%</td>
	<td data-value="17" class="abs low">2/17</td>
	</tr>

<tr>
	<td class="file low" data-value="MenuTitle.js"><a href="MenuTitle.js.html">MenuTitle.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	</tr>

<tr>
	<td class="file low" data-value="Modal.js"><a href="Modal.js.html">Modal.js</a></td>
	<td data-value="7.14" class="pic low"><div class="chart"><div class="cover-fill" style="width: 7%;"></div><div class="cover-empty" style="width:93%;"></div></div></td>
	<td data-value="7.14" class="pct low">7.14%</td>
	<td data-value="14" class="abs low">1/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="7.14" class="pct low">7.14%</td>
	<td data-value="14" class="abs low">1/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="MrmModal.js"><a href="MrmModal.js.html">MrmModal.js</a></td>
	<td data-value="57.14" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 57%;"></div><div class="cover-empty" style="width:43%;"></div></div></td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="7" class="abs medium">4/7</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="7" class="abs medium">4/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="Overlay.js"><a href="Overlay.js.html">Overlay.js</a></td>
	<td data-value="75" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 75%;"></div><div class="cover-empty" style="width:25%;"></div></div></td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	</tr>

<tr>
	<td class="file low" data-value="Pagination.js"><a href="Pagination.js.html">Pagination.js</a></td>
	<td data-value="9.68" class="pic low"><div class="chart"><div class="cover-fill" style="width: 9%;"></div><div class="cover-empty" style="width:91%;"></div></div></td>
	<td data-value="9.68" class="pct low">9.68%</td>
	<td data-value="31" class="abs low">3/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="10" class="pct low">10%</td>
	<td data-value="30" class="abs low">3/30</td>
	</tr>

<tr>
	<td class="file low" data-value="Placeholder.js"><a href="Placeholder.js.html">Placeholder.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	</tr>

<tr>
	<td class="file low" data-value="SelectImage.js"><a href="SelectImage.js.html">SelectImage.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	</tr>

<tr>
	<td class="file medium" data-value="SelectInput.js"><a href="SelectInput.js.html">SelectInput.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	</tr>

<tr>
	<td class="file low" data-value="Sort.js"><a href="Sort.js.html">Sort.js</a></td>
	<td data-value="16.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 16%;"></div><div class="cover-empty" style="width:84%;"></div></div></td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="12" class="abs low">2/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="12" class="abs low">2/12</td>
	</tr>

<tr>
	<td class="file high" data-value="Spinner.js"><a href="Spinner.js.html">Spinner.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file low" data-value="TheCrestInputs.js"><a href="TheCrestInputs.js.html">TheCrestInputs.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	</tr>

<tr>
	<td class="file low" data-value="Tooltip.js"><a href="Tooltip.js.html">Tooltip.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Apr 11 2019 19:58:40 GMT+0300 (East Africa Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
