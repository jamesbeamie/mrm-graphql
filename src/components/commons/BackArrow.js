import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import '../../assets/styles/backArrow.scss';

/**
 * Reusable component for a back arrow
 *
 * @param {object} redirectUri
 *
 * @returns {JSX}
 */

/* eslint-disable jsx-a11y/accessible-emoji */
const BackArrow = ({ redirectUri }) => (
  <NavLink to={redirectUri} className="arrow">
    <span className="arrow-symbol">&#11013;</span>
    <span>Back</span>
  </NavLink>
);

BackArrow.propTypes = {
  redirectUri: PropTypes.string.isRequired,
};

export default BackArrow;
