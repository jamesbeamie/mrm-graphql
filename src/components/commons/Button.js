import React from 'react';
import PropTypes from 'prop-types';

// styles
import '../../assets/styles/button.scss';

/**
 * It renders a button
 *
 * @param {Object}
 *
 * @returns {JSX}
 */
const Button = ({
  title,
  classProp,
  handleClick,
  isDisabled,
  type,
}) => {
  const classname = `${classProp} ${type === 2 ? 'btn-secondary' : 'btn-primary'}`;
  return (
    <button
      className={classname}
      onClick={handleClick}
      disabled={isDisabled}
    >
      <span>{title}</span>
    </button>
  );
};

Button.defaultProps = {
  classProp: null,
  handleClick: null,
  isDisabled: false,
  type: 1,
};

Button.propTypes = {
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
  classProp: PropTypes.string,
  handleClick: PropTypes.func,
  isDisabled: PropTypes.bool,
  type: PropTypes.number,
};

export default Button;
