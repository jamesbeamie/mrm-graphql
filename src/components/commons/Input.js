import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../../assets/styles/input.scss';
import Controls from '../helpers/Controls';

export class Input extends Component {
  constructor(props) {
    super(props);
    this.inputRef = React.createRef();
  }

  clear = () => {
    this.inputRef.current.value = 0;
  };

  /**
   * Increases and decreases numeric input value
   *
   * @param {object} event
   *
   * @returns {void}
   */
  handleIncrement = (event) => {
    const { onChange } = this.props;
    event.preventDefault();

    /* Logic for increasing and decreasing numeric input value
     * should be placed here.
     */

    const num = parseInt(this.inputRef.current.value, 10);
    if (event.target.name === 'up') {
      /* The button up event and Incremented number is passed
       * to the onChange function when the up button is
       * pressed.
       */
      onChange(event, num + 1);
      this.inputRef.current.value = num + 1;
    }
    if (event.target.name === 'down') {
      if (num >= 1) {
        onChange(event, num - 1);
        this.inputRef.current.value = num - 1;
      }
    }
  };

  render() {
    const {
      name,
      value,
      inputClass,
      type,
      labelName,
      onBlur,
      labelClass,
      id,
      placeholder,
      onChange,
      onFocus,
      controlsClass,
      ...otherProps
    } = this.props;

    return (
      <div className={labelClass}>
        <label htmlFor={name}>
          {labelName}
          <input
            ref={this.inputRef}
            type={type}
            className={inputClass}
            name={name}
            defaultValue={value}
            onChange={onChange}
            onBlur={onBlur}
            onFocus={onFocus}
            placeholder={placeholder}
            id={id}
            {...otherProps}
          />

          {type === 'number' ? (
            <Controls
              controlsClass={controlsClass}
              handleIncrement={this.handleIncrement}
            />
          ) : null}
        </label>
      </div>
    );
  }
}

Input.propTypes = {
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  name: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  inputClass: PropTypes.string,
  controlsClass: PropTypes.string,
  id: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  labelName: PropTypes.string.isRequired,
  labelClass: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  type: PropTypes.string,
};

Input.defaultProps = {
  value: 'text',
  type: 'text',
  onFocus: (() => {}),
  onBlur: (() => {}),
  inputClass: 'mrm-input default-input',
  labelClass: 'input1',
  controlsClass: '',
  placeholder: '',
  id: '',
};

export default Input;
