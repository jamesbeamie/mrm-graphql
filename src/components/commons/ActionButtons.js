import React from 'react';
import PropTypes from 'prop-types';
import Spinner from './Spinner';
import Button from '../commons/Button';

import '../../assets/styles/actionbuttons.scss';

/**
 * Reusable button component for performing
 * actions and canceling
 *
 * @param {array} props
 *
 * @returns {JSX}
 */
const ActionButtons = props => (
  props.isLoading
    ? <Spinner size="small" /> :
    <div className="form-action-buttons">
      <Button handleClick={props.onClickCancel} title={props.cancelButtonText} />
      { props.showActionButton &&
      <Button type={2} handleClick={props.onClickSubmit} title={props.actionButtonText} classProp="main-button" />
      }
    </div>
);

ActionButtons.propTypes = {
  onClickCancel: PropTypes.func.isRequired,
  cancelButtonText: PropTypes.string,
  actionButtonText: PropTypes.string,
  isLoading: PropTypes.bool,
  onClickSubmit: PropTypes.func,
  showActionButton: PropTypes.bool,
};

ActionButtons.defaultProps = {
  onClickSubmit: PropTypes.func,
  cancelButtonText: 'CANCEL',
  actionButtonText: 'SAVE CHANGES',
  isLoading: false,
  showActionButton: true,
};

export default ActionButtons;
