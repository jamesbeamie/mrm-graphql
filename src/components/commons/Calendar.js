import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import 'react-day-picker/lib/style.css';
import Button from '../commons/Button';
import PickRange from '../helpers/RangePicker';

// styles
import '../../assets/styles/calendar.scss';

moment.suppressDeprecationWarnings = true;

/**
 * Reusable component for Calendar
 *
 * @extends Component
 *
 * @returns {JSX}
 */
class Calendar extends Component {
  static propTypes = {
    sendData: PropTypes.func,
    classProp: PropTypes.string,
    startDate: PropTypes.string,
    endDate: PropTypes.string,
    disabledDateRange: PropTypes.string,
  };

  static defaultProps = {
    sendData: null,
    classProp: '',
    startDate: '',
    endDate: '',
    disabledDateRange: '',
  };

  state = {
    isCalendarOpen: false,
    startDate: this.props.startDate || moment().format('MMM DD Y'),
    endDate: this.props.endDate || moment().format('MMM DD Y'),
  };

  /**
   * Update the state of start and end dates
   *
   * @param {object} value
   *
   * @returns {void}
   */
  handleChange = (value) => {
    const dateTo = value.to.toString().slice(4, 15);
    const dateFrom = value.from.toString().slice(4, 15);
    this.setState({ startDate: dateFrom, endDate: dateTo || dateFrom });
  };

  /**
   * Sends the selected dates
   *
   * @returns {void}
   */
  sendDate = () => {
    const { startDate, endDate } = this.state;
    this.props.sendData(startDate, endDate);
    this.toggleCalendar();
  };

  /**
   * It toggles the calendar view
   *
   * @returns {void}
   */
  toggleCalendar = () => {
    this.setState({ isCalendarOpen: !this.state.isCalendarOpen });
  };

  render() {
    const { isCalendarOpen, startDate, endDate } = this.state;
    const { disabledDateRange } = this.props;
    return (
      <Fragment>
        <Button
          title={`${startDate} - ${endDate}`}
          classProp={`calendarIcon ${this.props.classProp}`}
          type={2}
          handleClick={this.toggleCalendar}
        />
        {isCalendarOpen && (
          <div className="calendar">
            <PickRange handleChange={this.handleChange} disabledDateRange={disabledDateRange} />
            <div className="calendar__button">
              <button
                onClick={this.toggleCalendar}
                className="calendar__button__item cancel_button"
              >
                Cancel
              </button>
              <button
                type="button"
                onClick={() => this.sendDate()}
                className="calendar__button__item apply_button"
              >
                Apply
              </button>
            </div>
          </div>
        )}
      </Fragment>
    );
  }
}

export default Calendar;
