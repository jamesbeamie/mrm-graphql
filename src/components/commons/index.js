import SelectInput from './SelectInput';
import Modal from './Modal';
import { Input } from './Input';
import ActionButtons from './ActionButtons';
import IconButtons from './IconButtons';

export { SelectInput, Modal, Input, ActionButtons, IconButtons };
