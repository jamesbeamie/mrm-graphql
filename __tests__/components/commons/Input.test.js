import React from 'react';
import { shallow, mount } from 'enzyme';
import { Input } from '../../../src/components/commons/Input';

describe('Input Component', () => {
  const props = {
    name: '',
    value: '',
    inputClass: '',
    id: '',
    labelName: '',
    labelClass: '',
    onChange: jest.fn(),
  };
  const wrapper = shallow(<Input {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have one input', () => {
    const modalForm = wrapper.find('input');
    expect(modalForm).toHaveLength(1);
  });

  it('should have one label', () => {
    const modalForm = wrapper.find('label');
    expect(modalForm).toHaveLength(1);
  });
});

describe('Input mount', () => {
  const props = {
    name: 'numberField',
    value: 1,
    inputClass: '',
    id: '',
    type: 'number',
    labelName: '',
    labelClass: '',
    onChange: jest.fn(),
  };
  const wrapper = mount(<Input {...props} />);
  it('should decerease on button down click', () => {
    wrapper.find('input').instance().value = 2;
    wrapper.find('button.down').simulate('click', { target: { name: 'down' } });
    expect(wrapper.find('input').instance().value).toEqual('1');
  });

  it('should not decerease on button down click', () => {
    wrapper.find('input').instance().value = 0;
    wrapper.find('button.down').simulate('click', { target: { name: 'down' } });
    expect(wrapper.find('input').instance().value).toEqual('0');
  });

  it('should increase on button up click', () => {
    wrapper.instance().clear();
    const spy = jest.spyOn(wrapper.instance(), 'handleIncrement');
    wrapper.instance().forceUpdate();
    wrapper.find('button.up').simulate('click', { target: { name: 'up' } });
    expect(spy).toHaveBeenCalled();
  });
});
