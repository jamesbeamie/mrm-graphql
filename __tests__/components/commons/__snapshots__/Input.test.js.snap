// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders properly 1`] = `
<div
  className=""
>
  <label
    htmlFor=""
  >
    <input
      className=""
      defaultValue=""
      id=""
      name=""
      onBlur={[Function]}
      onChange={[MockFunction]}
      onFocus={[Function]}
      placeholder=""
      type="text"
    />
  </label>
</div>
`;
